### DISCUSSION TITLE
Can we use @cloudflare/puppeteer in an edge function?

### DISCUSSION THREAD

**Discussion by mattiaz9** [vercel/community#3017] [Idea]
Since edge functions are wrappers on top of cloudflare workers I'm wondering if they can support puppeteer?

According to this page it should be supported in a worker, the question is if Vercel supports it.
https://developers.cloudflare.com/browser-rendering/platform/puppeteer/#use-puppeteer-in-a-worker

**Comment by smaeda-ks** [comment]
No, we don't support it as of today. Would you mind sharing your use case?

**Reply by mattiaz9** [reply]
generating a pdf from some html

**Reply by smaeda-ks** [reply]
I see. As far as I know, https://github.com/cheeriojs/cheerio works with Edge Runtime

Generating PDF might still be tricky but can be done in serverless using https://react-pdf.org/ for example

**Reply by mattiaz9** [reply]
None of those can work for my needs.

Anyway, I'll probably deploy a Cloudflare Worker in the meantime. Hopefully in the future Vercel is going to support it.

**Reply by lukkoro** [reply]
@mattiaz9 Did you manage to make the `@cloudflare/puppeteer` work in an Edge runtime? I tried replacing the puppeteer library with the Cloudflare one, but it turns out when I change the runtime to edge, the base package uses node libraries...

I am trying to deploy Next.js app on Cloudflare using the next-on-pages package, and all functions need to be in an edge runtime, but puppeteer lib is complaining

**Reply by mattiaz9** [reply]
no, as @smaeda-ks said, it is not supported. 
I haven't played with next.js on Cloudflare, but to use the browser rendering api you need to enable the `nodejs_compat` flag. I don't know if that's possible from next.js.
Also you will need to get access to the browser-rendering-api-beta to use it anyway.
